### PR TITLE
[Feature] Summary

### PR DESCRIPTION
**Pull request by KobbyMmo**
## Description

Create a text Block with a column structure

Fixes # ([179117122](https://www.pivotaltracker.com/story/show/179117122))

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image

image

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
I must be having one of those slow days @KobbyMmo so bare with me... But I don't get what exactly what are we doing here. A bit of context would be appreciated.

**Comment by KobbyMmo** [comment]
> I must be having one of those slow days @KobbyMmo so bare with me... But I don't get what exactly what are we doing here. A bit of context would be appreciated.

So the ticket requested we create a 2 column layout to show the text block here which appears in about and how it works. But I believe, Gutenberg already provides a good column structure, So I added classes to style it directly from the cms instead of using blocks. @kilemensi 
image

**Comment by kilemensi** [comment]
> So the ticket requested we create a 2 column ...
>

Cool cool @KobbyMmo. To confirm, are we able to limit layout, the number of columns, etc. in Gutenberg?

**Comment by KhadijaMahanga** [comment]
@kilemensi not sure we can. The content team has a room to add anything/any block on that text lazy-block.

Am I right @KobbyMmo?

**Comment by KobbyMmo** [comment]
> @kilemensi not sure we can. The content team has a room to add anything/any block on that text lazy-block.

No we can't @KhadijaMahanga

**Comment by kilemensi** [comment]
Any update on this @KobbyMmo ?

**Comment by KhadijaMahanga** [comment]
@KobbyMmo ?

File: src/functions/formatBlockData.js [hunk 1]
```diff
@@ -34,7 +33,12 @@ export default async function formatBlockData(blocks) {
...
-      return { name, attributes, innerBlocks: innerBlocksFormatted };
+      return {
+        name,
+        attributes,
+        content: saveContent,
+        innerBlocks: innerBlocksFormatted,
+      };
     })
   );
 }
```
Comment (KhadijaMahanga) [inline src/functions/formatBlockData.js:44]: Do we still need changes in here? 

I went to a conclusion that we don't need this function at all.

File: src/pages/how-it-works.js [hunk 2]
```diff
@@ -3,20 +3,23 @@ import React from "react";
...
 import Page from "@/pesayetu/components/Page";
+import TwoColumnLayout from "@/pesayetu/components/TwoColumnLayout";
 import formatBlocksForSections from "@/pesayetu/functions/formatBlocksForSections";
 import getPostTypeStaticProps from "@/pesayetu/functions/postTypes/getPostTypeStaticProps";
 
 export default function Home({ blocks, ...props }) {
   return (
     <Page {...props}>
       <Hero {...blocks?.otherHero} />
+      <TwoColumnLayout {...blocks.twoColumnLayout} />
```
Comment (KhadijaMahanga) [inline src/pages/how-it-works.js:14]:
```suggestion
      <TwoColumnLayout {...blocks?.twoColumnLayout} />
```

**Review by KhadijaMahanga — COMMENTED** [review]
@KobbyMmo this component appears on how-it-works page too. Let's add it.

And title & subtitle font color doesn't match design

**Review by KhadijaMahanga — APPROVED** [review]
🚀

File: src/components/TwoColumnLayout/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        <Grid container>
+          <Grid item xs={12} lg={4}>
+            <Typography className={classes.title} variant="h1">
+              {title}
+            </Typography>
+            <Typography className={classes.subtitle} variant="h4">
+              {subtitle}
+            </Typography>
+          </Grid>
+          <Grid item lg={2} />
```
Comment (kilemensi) [inline src/components/TwoColumnLayout/index.js:28]: If we need to skip columns **_between_** two `Grid item`s, wouldn't justifyContent="space-between" be enough?

File: src/components/TwoColumnLayout/useStyles.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      objectFit: "contain",
+      height: "auto",
+    },
+    "& .wp-block-image": {
+      width: "100%",
+      maxWidth: "100%",
+      margin: 0,
+      marginTop: typography.pxToRem(20),
+      marginBottom: typography.pxToRem(20),
+    },
```
Comment (kilemensi) [inline src/components/TwoColumnLayout/useStyles.js:43]: Can you show an example of how images would look?

**Review by kilemensi — COMMENTED** [review]
👍🏾  

---------

Lets call the component `History`; Calling it `TwoColumnLayout` implies it is generic enough to be used anywhere we need a two-column layout.

**Comment by KobbyMmo** [comment]
> Lets call the component `History`; Calling it `TwoColumnLayout` implies it is generic enough to be used anywhere we need a two-column layout.

We use it for `Methodology` as well @kilemensi

**Comment by kilemensi** [comment]
> > Lets call the component `History`; Calling it `TwoColumnLayout` implies it is generic enough to be used anywhere we need a two-column layout.
> 
> We use it for `Methodology` as well @kilemensi

Cool, find a better name then @KobbyMmo

File: src/components/TwoColumnLayout/useStyles.js (hunk 4 above)
Comment (KobbyMmo) [reply src/components/TwoColumnLayout/useStyles.js]: image
Here @kilemensi

File: src/components/MetaTextBlock/useStyles.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  section: {
+    paddingBottom: typography.pxToRem(80),
+    paddingTop: typography.pxToRem(80),
+  },
+  title: {
+    fontSize: typography.pxToRem(50),
+    lineHeight: 66 / 50,
+    marginBottom: typography.pxToRem(40),
+    fontWeight: "300",
```
Comment (kilemensi) [inline src/components/MetaTextBlock/useStyles.js:15]: Have you checked that these font settings are not defined in theme?

File: src/components/MetaTextBlock/index.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  if (!content) {
+    return null;
+  }
+
+  return (
+    <div className={classes.root}>
+      <Section classes={{ root: classes.section }}>
+        <Grid justifyContent="space-between" container>
+          <Grid item xs={12} lg={4}>
+            <Typography className={classes.title} variant="h1">
```
Comment (kilemensi) [inline src/components/MetaTextBlock/index.js:21]:
```suggestion
            <Typography className={classes.title} variant="h2">
```

https://github.com/CodeForAfrica/PesaYetu/blob/b0c9a40839e9df6e0ca1cea46906420457d32794/src/theme/index.js#L127-L131

File: src/pages/about.js [hunk 7]
```diff
@@ -2,29 +2,29 @@ import PropTypes from "prop-types";
...
-import getFooterMenu from "@/pesayetu/functions/menus/getFooterMenu";
 import getPostTypeStaticProps from "@/pesayetu/functions/postTypes/getPostTypeStaticProps";
 
-export default function About({ footerProps, blocks, ...props }) {
+export default function About({ blocks, ...props }) {
   return (
-    <Page footerProps={footerProps} {...props}>
+    <Page {...props}>
       <Hero {...blocks?.aboutHero} />
+      <MetaTextBlock {...blocks?.twoColumnLayout} />
```
Comment (kilemensi) [inline src/pages/about.js:14]: What the hell is `MetaTextBlock`?

**Review by kilemensi — COMMENTED** [review]
👍🏾 

---

So what's the new name?

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**